src/umbraco.cms/Actions/ActionRestore.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using umbraco.interfaces;
+using umbraco.BasePages;
+
+namespace umbraco.BusinessLogic.Actions
+{
+	/// <summary>
+	/// This action is invoked when the content item is to be restored from the recycle bin
+	/// </summary>
+	public class ActionRestore : IAction
+	{
+		//create singleton
+#pragma warning disable 612,618
+		private static readonly ActionRestore m_instance = new ActionRestore();
```

Comment:
Yeah, the warning was because of the `Obsolete` attribute on the constructor... which again, was because I copied from similar pattern of other `IAction` implementations. Thanks for fixing it up!